src/main/java/com/oltpbenchmark/api/Worker.java

```diff
@@ -430,6 +446,14 @@ protected final void doWork(DatabaseType databaseType, TransactionType transacti
                     }
 
                 } finally {
+                    if (this.configuration.getNewConnectionPerTxn() && this.conn != null) {
+                        try {
+                            this.conn.close();
+                            this.conn = null;
+                        } catch (SQLException e) {
+                            LOG.error("Connection couldn't be closed.", e);
```

Comment:
When would this happen? This seems like it should be a fatal error?

Reply:
I just duplicated the behavior in `tearDown()`. Presumably the `conn.close()` call can throw which is why the logic is the way it is in `tearDown()`.